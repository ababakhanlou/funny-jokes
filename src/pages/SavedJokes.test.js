import React from "react";
import { shallow } from "enzyme";
import { localStorageMock } from "../testHelpers/mockLocalStorage";

describe("SavedJokes", () => {
  let wrapper;
  const mockReload = jest.fn();

  Object.defineProperty(window, "localStorage", { value: localStorageMock });
  Object.defineProperty(window, "location", { value: { reload: mockReload } });

  jest.mock("../components", () => ({
    Card: () => <div>Card</div>,
  }));

  const SavedJokes = require("./SavedJokes").default;

  wrapper = shallow(<SavedJokes />);

  it("should render correctly with no jokes", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should render correctly with list of jokes", () => {
    localStorageMock.setItem("jokes", "A Joke 1<#>A Joke 2<#>");
    wrapper = shallow(<SavedJokes />);

    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should delete a joke on delete click", () => {
    const deleteJoke = wrapper.find("Card").at(1).prop("action");
    deleteJoke();
    expect(localStorageMock.getItem("jokes")).toBe("A Joke 1");
  });
});
