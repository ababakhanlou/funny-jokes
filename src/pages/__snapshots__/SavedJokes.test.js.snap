// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedJokes should render correctly with list of jokes 1`] = `
"<div className=\\"App-jokeList\\">
  <Card joke=\\"A Joke 1\\" button={true} action={[Function: action]} />
  <Card joke=\\"A Joke 2\\" button={true} action={[Function: action]} />
</div>"
`;

exports[`SavedJokes should render correctly with no jokes 1`] = `
"<div className=\\"App-jokeList\\">
  <p>
    We have no jokes to serve you
  </p>
</div>"
`;
